lib/ansible/plugins/connection/ssh.py

```diff
@@ -347,7 +352,17 @@ def wrapped(self, *args, **kwargs):
                 if return_tuple[0] != 255:
                     break
                 else:
+                    SSH_ERROR = True
+                    for signature in NOT_SSH_ERRORS:
```

Comment:
One liners are less readable than conditionals.  In general we should prefer to use loops and if (sivel and I commented as much on one of your PRs a few weeks ago IIRC).  This is also less efficient in general as it has to process every element in the data instead of breaking early.  (Might be a wash here as there's only two elements and we're trading a loop in Python for a loop in C).

list comprehensions and generator expressions are good for list processing tasks.  It's when they are used to combine loops and conditions that the code starts to smell (or if the amount of logic shoved into the list comprehension becomes too large... although I haven't figured out what the value of "too alre" should be yet).

Reply:
FTR since `all()` accepts a generator expression it breaks early :)